jupyterhub_moss/spawner.py

```diff
@@ -213,12 +214,22 @@ def options_from_form(self, formdata: Dict[str, List[str]]) -> Dict[str, str]:
                 raise RuntimeError("GPU(s) not available for this partition")
             options["gres"] = gpu_gres_template.format(options["ngpus"])
 
+        partition_environments = tuple(
+            self.partitions[partition]["jupyter_environments"].values()
+        )
         if "environment_path" not in options:
             # Set path to use from first environment for the current partition
-            default_venv = tuple(
-                self.partitions[partition]["jupyter_environments"].values()
-            )[0]
-            options["environment_path"] = default_venv["path"]
+            options["environment_path"] = partition_environments[0]["path"]
+
+        matching_envs_without_add_to_path = [
+            env
+            for env in partition_environments
+            if env["path"] == options["environment_path"]
+            and not env.get("add_to_path", True)
+        ]
+        if not matching_envs_without_add_to_path:
```

Comment:
I replaced the `for .. else` pattern with a list filter and a test of list content, hope that's simpler to read.

Reply:
BTW the alternative to looping through the list is to use the path as the key for the jupyter envs and put the key as an `id` item of the dict.